docs/contributing/999-rfcs/2023-10-01-workloads.md

```diff
@@ -0,0 +1,262 @@
+---
+title: "#4350 Containerized Workloads - Requirements"
+description: Requirements for containerized workloads in the Wing Cloud Library
+---
+
+# Containerized Workloads - Requirements Specification
+
+- **Author(s)**: @eladb
+- **Submission Date**: 2023-10-01
+- **Stage**: Proposal
+- **Stage Date**: 2023-10-01
+
+This specification lists the *requirements* we have identified so far that are related to using
+containers to implement long running workloads (aka "services" or "microservices"). 
+
+## Requirements
+
+Requirements are marked with a *Reqtag* so they will be easy to trace in subsequent design
+documents. Suggested priorities are marked inline as "(P1)", "(P2)" or "(P3)".
+
+### Containerized workloads (P1)
+
+> Reqtag: `c:workload`
+
+A "workload" describes a unit of scalable deployment. It's akin to a K8S's deployment + service + ingress, or ECS's task or fly.io's machine or ControlPlane's workload.
+Wing must support defining **containerized workloads** implemented through user-defined container
+images. Images may either be pulled from a container registry or built from a `Dockerfile` within
+the project directory.
+
+### Simulator implementation (P1)
+
+> Reqtag: `c:simulator`
+
+When using Docker images, the Wing simulator will need to take care of seamlessly building and/or
+pulling the container image and maintaining it's lifecycle and cleanup via a local docker
+installation.
+
+If docker is not installed, an error should provide useful instructions on how to install it.
+
+### Hot reloading for fast local iteration (P1)
+
+> Reqtag: `c:hot-reloading`
+
+When developing a Wing application which references a locally defined container image, we would like
+to support live updates which to instantly reload the program when the source changes.
```

Comment:
> when the source changes

Is this a reference to the source code thats bundled in the container, the Dockerfile, or the local image tag? Maybe its a combination, just wanted to clarify how this hot-reload picture looks.